tests/custom_messages_worker_shell.html

```diff
@@ -0,0 +1,86 @@
+    <title>Custom messaging worker test</title>
+  </head>
+  <body>
+     <div>
+      <canvas id="canvas"></canvas>
+    </div>
+    <textarea id="output" rows="8"></textarea>
+    <script type="text/javascript">
+      function sendResult(success) {
+        if (window.sentTestResult) return;
+        window.sentTestResult = true;
+        var xhr = new XMLHttpRequest();
+        xhr.open('GET', 'http://localhost:8888/report_result?' + (success ? 1 : 0));
+        xhr.send();
+        setTimeout(function() { window.close() }, 1000);
+      }
+
+      var Module = {
+        print: (function() {
+          var element = document.getElementById('output');
+          if (element) element.value = ''; // clear browser cache
+          return function(text) {
+            console.log(text);
+            if (element) {
+              element.value += text + "\n";
+              element.scrollTop = element.scrollHeight; // focus on bottom
+            }
+          };
+        })(),
+        printErr: function(text) {
+          if (arguments.length > 1) text = Array.prototype.slice.call(arguments).join(' ');
+          if (0) { // XXX disabled for safety typeof dump == 'function') {
+            dump(text + '\n'); // fast, straight to the real console
+          } else {
+            console.error(text);
+          }
+        },
+        canvas: (function() {
+          var canvas = document.getElementById('canvas');
+          // As a default initial behavior, pop up an alert when webgl context is lost. To make your
+          // application robust, you may want to override this behavior before shipping!
+          // See http://www.khronos.org/registry/webgl/specs/latest/1.0/#5.15.2
+          canvas.addEventListener("webglcontextlost", function(e) { alert('WebGL context lost. You will need to reload the page.'); e.preventDefault(); }, false);
+
+          return canvas;
+        })(),
+
+        onCustomMessage: function(message) {
+          function replyToWorker(msg, cmd) {
+            let w = Module.PThread.pthreads[msg.worker].worker;
+            w.postMessage({
+                cmd: 'custom',
+                customCmd: cmd,
+                payload: msg.payload + '[' + cmd + ']'
+            });
+          }
+
+          switch (message.customCmd) {
+            case 'first_msg': {
+              replyToWorker(message, 'reply_first');
+              break;
+            }
+            case 'second_msg': {
+              replyToWorker(message, 'reply_second');
+              break;
+            }
+            case 'last_msg': {
+              sendResult(message.payload === '[first][reply_first][second][reply_second][last]');
+              break;
+            }
+            default: {
+              Module.print('unknown custom message');
+              sendResult(false);
+            }
+          }
+        }
+      };
+    </script>
+    {{{ SCRIPT }}}
+  </body>
+</html>
```

Comment:
I don't think we should need to add this html file .. I think we can just set Module.onCustomMessage in a pre-js, no?  (that way the test will run on node too).

Reply:
Good point. I got that working for a browser test, but I suppose it shouldn't be a browser test if it's to be ran with node?

Is core OK or should it be a 'other' test? (Or something else entirely?)